File: declined-python/declined/resources/analytics.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any

from declined.paths import API_PATHS

if TYPE_CHECKING:
    from declined.client import HttpTransport


class AnalyticsResource:
    def __init__(self, http: HttpTransport):
        self._http = http

    def get(self, **params: Any) -> Any:
        return self._http.request("GET", API_PATHS["analytics"], params=params or None)

File: declined-node/src/resources/analytics.ts
import type { HttpClient } from "../client.js";
import { API_PATHS } from "../paths.js";
import type { ListParams } from "../types.js";

export class AnalyticsResource {
  constructor(private readonly http: HttpClient) {}

  get(params?: ListParams) {
    return this.http.request("GET", API_PATHS.analytics, undefined, params);
  }
}
